runtime/node/azurewebapp/src/index.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as restify from 'restify';
+import * as fs from 'fs';
+import * as path from 'path';
+import { BotFrameworkAdapter, MemoryStorage, ConversationState, UserState } from 'botbuilder';
+import { DialogManager } from 'botbuilder-dialogs';
+import { AdaptiveDialog, AdaptiveDialogComponentRegistration, LanguageGeneratorMiddleWare } from 'botbuilder-dialogs-adaptive';
+import { ResourceExplorer } from 'botbuilder-dialogs-declarative';
+
+// Create HTTP server.
+const server = restify.createServer();
+const argv = require('minimist')(process.argv.slice(2));
+server.listen(process.env.port || process.env.PORT || argv.port || 3978, (): void => {
+    console.log(`\nGet Bot Framework Emulator: https://aka.ms/botframework-emulator`);
+    console.log(`\nTo talk to your bot, open echobot.bot file in the Emulator.`);
+});
+
+// Load project settings
+let projectSettings = {
+    bot: '../../',
+    root: '../../'
+};
+if (process.env.node_environment === 'production') {
+    projectSettings = require('../appsettings.deployment.json');
+} else {
+    projectSettings = require('../appsettings.development.json');
+}
+
+const projectRoot = path.join(__dirname, '../', projectSettings.root);
+
+// Find entry dialog file
+let mainDialog = 'main.dialog';
+const files = fs.readdirSync(projectRoot);
+for (let file of files) {
+    if (file.endsWith('.dialog')) {
+        mainDialog = file;
+        break;
+    }
+}
+
+// Create resource explorer.
+const resourceExplorer = new ResourceExplorer().addFolders(projectRoot, ['runtime'], false)
+resourceExplorer.addComponent(new AdaptiveDialogComponentRegistration(resourceExplorer));
+
+// Create adapter.
+// See https://aka.ms/about-bot-adapter to learn more about .bot file its use and bot configuration.
+const adapter = new BotFrameworkAdapter({
+    appId: process.env.microsoftAppID,// || '97f8a0c9-93d0-4bad-a3e3-30b1aaf79aa0',
+    appPassword: process.env.microsoftAppPassword,// || '4ve=H:PYD?.NXn9jHORhW8e9mr9SSnm_',
+});
+adapter.use(new LanguageGeneratorMiddleWare(resourceExplorer));
+
+const bot = new DialogManager();
+bot.userState = new UserState(new MemoryStorage());
```

Comment:
we will need to apply all the same configurable options to this bot as we did for C# - that is, ability to swap out storage providers, turn on middlewares, etc.